chore: improve pull request template

HonkingGoose opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
<!-- Your pull request title must be a Conventional Commits style message (https://www.conventionalcommits.org/en/v1.0.0/) -->
<!-- If your pull request is going to change a lot of things, open an issue for discussion before starting your work. -->

## Changes:

<!-- Describe what this pull request changes. -->

- Use active voice
- Simplify text
- Be clear when things are required from the contributor, I think the old pull request is a bit too hand wavy on hard requirements. This is a tone/style thing that affects the way the Renovate project feels to newcomers, so feel free to adjust this with a review. :wink: 
- Add proper headings for sections
- Add documentation and testing check-boxes

## Context:

<!-- Describe why you're making these changes. -->
<!-- If you're closing an existing issue with this pull request, use the keyboard Closes #issue_number -->

The pull request template needs to be updated.
This pull request template incorporates the good stuff from the discussion at issue #5189.

Closes #5189 

## Documentation (please tick one)

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [X] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

HonkingGoose commented:
By the way, I'm using the new template here. So you can see how it looks. :smile:

viceice reviewed (approved):
like it

rarkins commented on .github/pull_request_template.md:
```diff
@@ -1,9 +1,26 @@
+## Context:
 
-<!-- Finish the PR with `Closes #` and then the issue number if it closes any associated issue -->
+<!-- Describe why you're making these changes. -->
```
```suggestion
<!-- Describe why you're making these changes if it's not already explained in a corresponding issue. -->
```

rarkins commented on .github/pull_request_template.md:
```diff
@@ -1,9 +1,26 @@
 
-<!-- Finish the PR with `Closes #` and then the issue number if it closes any associated issue -->
+<!-- Describe why you're making these changes. -->
+<!-- If you're closing an existing issue with this pull request, use the keyboard Closes #issue_number -->
```
```suggestion
<!-- If you're closing an existing issue with this pull request, use the keyword Closes #issue_number -->
```

HonkingGoose replied:
Good catch, I don't know why I typed that.... :smile:

rarkins commented on .github/pull_request_template.md:
```diff
@@ -1,9 +1,26 @@
+<!-- Describe why you're making these changes. -->
+<!-- If you're closing an existing issue with this pull request, use the keyboard Closes #issue_number -->
+
+## Documentation (please tick one)
```
```suggestion
## Documentation (please check one with an [x])
```

rarkins commented:
I think the conventional commits reference is a bit "heavy" and could put people off who aren't familiar with it. I suggest either:
- We create our own very short description with examples to link to, OR
- Leave it off because it's really easy for any maintainer to edit the PR title once it's created anyway

HonkingGoose commented:
I've removed the reference to the Conventional Commits style guide. Less is more. :smile:

rarkins commented on .github/pull_request_template.md:
```diff
@@ -1,6 +1,5 @@
 <!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
 <!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
-<!-- Your pull request title must be a Conventional Commits style message (https://www.conventionalcommits.org/en/v1.0.0/) -->
 <!-- If your pull request is going to change a lot of things, open an issue for discussion before starting your work. -->
```
I have a weird suggestion. What if we remove this line too?

Reasoning:
- The benefit of this guidance is 80-90% the contributor's, and only 10-20% maintainers. e.g. it takes a little time for us to review a PR based on a bad idea and say sorry but this is not really what we want.. but most of the time wasted is theirs
- The only time someone is likely to see this template is when they have already pushed a branch and clicked "Create Pull Request".. i.e. they've already done the work
- Could this message discourage people who would otherwise submit a useful PR?
- Can we cover it in `CONTRIBUTING.md` instead?

HonkingGoose replied:
I've removed this line:

```markdown
<!-- If your pull request is going to change a lot of things, open an issue for discussion before starting your work. -->
```

rarkins commented on .github/pull_request_template.md:
```diff
@@ -1,9 +1,25 @@
+
+- [ ] Code inspection only, or
+- [ ] Newly added unit tests, or
+- [ ] Unit tests + ran on a real repository
```
```suggestion
- [ ] Unit tests + ran on a real repository


<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/master/.github/pull_request_template.md -->
```

HonkingGoose commented:
@rarkins I've implemented your review feedback, fixed the small Prettier linting issue that your suggestion caused, and merged the current `master` branch.

I think this is ready for another round of review. :smile:
